Community/Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js

```diff
@@ -0,0 +1,400 @@
+        // bitrate again and since this can be inaccurate (It calculates overall bitrate not video specific)
+        // we have to inflate the current bitrate so we don't keep looping this logic.
+      } else if (inputs.reconvert_hevc === 'true' && (file.ffProbeData.streams[i].codec_name === 'hevc'
+        || file.ffProbeData.streams[i].codec_name === 'vp9' || file.ffProbeData.streams[i].codec_name === 'av1')) {
+        // If we're using the hevc max bitrate then update the cutoff to use it
+        if (inputs.hevc_max_bitrate !== '') {
+          if (currentBitrate > inputs.hevc_max_bitrate) {
+            inflatedCutoff = Math.round(inputs.bitrate_cutoff);
+            response.infoLog += `☒ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+            Using hevc specific cutoff of ${inputs.hevc_max_bitrate}. 
+            \n☒ The file is still above this new cutoff! Reconverting. \n`;
+          } else {
+            response.processFile = false;
+            inflatedCutoff = Math.round(inputs.bitrate_cutoff);
+            response.infoLog += `☑ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+            Using hevc specific cutoff of ${inputs.hevc_max_bitrate}. 
+            \n☑ The file is NOT above this new cutoff. Exiting plugin. \n`;
+            return response;
+          }
+          // If we're not using the hevc max bitrate then we need a safety net
+          // to try and ensure we don't keep looping this plugin.
+        } else if (currentBitrate > (inputs.bitrate_cutoff * 2)) {
+          inflatedCutoff = Math.round((inputs.bitrate_cutoff * 2));
+          response.infoLog += `☒ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1. 
+          hevc specific cutoff not set so bitrate_cutoff is multiplied by 2 for safety! 
+          Cutoff now temporarily ${inflatedCutoff}. \n The file is still above this new cutoff! Reconverting. \n\n`;
+        } else {
+          response.processFile = false;
+          inflatedCutoff = Math.round((inputs.bitrate_cutoff * 2));
+          response.infoLog += `☑ Reconvert_hevc is ${inputs.reconvert_hevc} & the file is already hevc, vp9 or av1 
+          so bitrate_cutoff is multiplied by 2! Cutoff now temporarily ${inflatedCutoff}. 
+          \n The file is NOT above this new cutoff. Exiting plugin. \n\n`;
+          return response;
+        }
+      }
+    }
+
+    // Are we encoding 10 bit files?
+    if (inputs.enable_10bit === 'true') {
+      // Check if video stream is 10bit. Currently set profile main10 and pixel format.
+      // Still need to test if this really works correctly
+      if (file.ffProbeData.streams[i].profile === 'High 10'
+        || file.ffProbeData.streams[i].bits_per_raw_sample === '10') {
+        main10 = '-profile:v main10 -pix_fmt p010le';
+        response.infoLog += '☑ File is 10bit or HDR. Setting Main10 Profile & pixel format \n\n';
+      }
+    }
```

Comment:
What if people want to convert from 8 bit to 10 bit? If you don't set these afaik it'll just copy the data from the source. HDR will get lost this way as the pixel format is different.

Reply:
Good catch. Overhauled this now. 
If this is set to true then we convert to 10 bit. Then the rest of the check is just for catching if the file is already 10 bit where we just disable hardware decode to avoid issues.
As for HDR. That now has a separate check which if found we then exit the plugin. HDR seems to be a bit involved and outside the scope of this plugin tbh.